File: C++/226_Invert_Binary_Tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        TreeNode *tmp;
        if(root)
        {
            tmp = root->left;
            root->left = invertTree(root->right);
            root->right = invertTree(tmp);
            return root;
        }else
        {
            return NULL;
        }

    }
};

File: Python/226_Invert_Binary_Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


#recursive
class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        if root:
            root.left,root.right = self.invertTree(root.right),self.invertTree(root.left)
            return root


#iterative
class Solution:
    def invertTree(self, root: TreeNode) -> TreeNode:
        
        stack = [root] if root else []
        
        while stack:
            node = stack.pop()
            if node:
                node.left,node.right = node.right,node.left
                stack.extend([node.left,node.right])
        return root
